src/frontend/src/flows/manage/index.ts

```diff
@@ -204,31 +210,25 @@ export const displayManage = (
   devices_: DeviceData[]
 ): Promise<void | AuthenticatedConnection> =>
   new Promise((resolve) => {
-    const hasSingleDevice = devices_.length <= 1;
-
-    const devices: Device[] = devices_.map((device) => {
-      return {
-        settings: deviceSettings({
-          userNumber,
-          connection,
-          device,
-          isOnlyDevice: hasSingleDevice,
-          reload: (newConnection?: AuthenticatedConnection) =>
-            resolve(newConnection),
-        }),
-        label: isRecoveryDevice(device)
-          ? recoveryDeviceToLabel(device)
-          : device.alias,
-        isRecovery: isRecoveryDevice(device),
-        isProtected: isProtected(device),
-        warn: domainWarning(device),
-      };
+    const devices = devicesFromDeviceDatas({
+      devices: devices_,
+      userNumber,
+      connection,
+      reload: resolve,
     });
-
+    if (devices.dupPhrase) {
+      toast.error(
+        "More than one recovery phrases are registered, which is unexpected. Only one will be showed."
+      );
+    }
```

Comment:
just in case, it's correct, it should not return after the error?

Reply:
(thanks for double checking though)